common/src/main/java/com/wynntils/screens/itemsharing/ItemSharingScreen.java

```diff
@@ -0,0 +1,225 @@
+
+        renderPreview(guiGraphics);
+    }
+
+    @Override
+    public void renderBackground(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTick) {
+        super.renderBackground(guiGraphics, mouseX, mouseY, partialTick);
+
+        RenderUtils.drawTexturedRect(guiGraphics.pose(), Texture.ITEM_SHARING_BACKGROUND, backgroundX, backgroundY);
+    }
+
+    private void renderPreview(GuiGraphics guiGraphics) {
+        guiGraphics.renderTooltip(
+                FontRenderer.getInstance().getFont(),
+                previewItemStack,
+                tooltipX,
+                backgroundY + (FontRenderer.getInstance().getFont().lineHeight * 2));
+    }
+
+    private void refreshPreview() {
+        // Encode the item with the selected settings
+        EncodingSettings encodingSettings = new EncodingSettings(
+                Models.ItemEncoding.extendedIdentificationEncoding.get(), Models.ItemEncoding.shareItemName.get());
+        ErrorOr<EncodedByteBuffer> errorOrEncodedByteBuffer =
+                Models.ItemEncoding.encodeItem(wynnItem, encodingSettings);
+        if (errorOrEncodedByteBuffer.hasError()) {
+            WynntilsMod.error("Failed to encode item: " + errorOrEncodedByteBuffer.getError());
+            McUtils.sendErrorToClient(I18n.get("feature.wynntils.chatItem.chatItemErrorEncode"));
+            McUtils.mc().setScreen(null);
+            return;
+        }
+
+        encodedItem = errorOrEncodedByteBuffer.getValue();
+
+        // Decode the item to be displayed
+        ErrorOr<WynnItem> errorOrDecodedByteBuffer = Models.ItemEncoding.decodeItem(encodedItem);
+        if (errorOrDecodedByteBuffer.hasError()) {
+            WynntilsMod.error("Failed to decode item: " + errorOrDecodedByteBuffer.getError());
+            McUtils.sendErrorToClient(I18n.get("feature.wynntils.chatItem.chatItemErrorDecode"));
+            McUtils.mc().setScreen(null);
+        }
+
+        WynnItem renderedItem = errorOrDecodedByteBuffer.getValue();
+
+        previewItemStack = new FakeItemStack(renderedItem, "From chat");
+
+        // Find the width of the tooltip
+        int tooltipWidth = LoreUtils.getTooltipLines(previewItemStack).stream()
+                .map(component -> McUtils.mc().font.width(component))
+                .max(Integer::compareTo)
+                .orElse(0);
+
+        // Total area the tooltip + sharing options panel + gap should cover
+        int totalRenderWidth = tooltipWidth + Texture.ITEM_SHARING_BACKGROUND.width() + 70;
+
+        // Blank space to the side of each element
+        int sideGap = (this.width - totalRenderWidth) / 2;
+
+        tooltipX = this.width - sideGap - tooltipWidth - 30;
+        backgroundX = sideGap + 30;
+    }
+
+    private void shareItem(String target) {
+        if (WynntilsMod.isDevelopmentEnvironment()) {
+            WynntilsMod.info("Encoded item: " + encodedItem);
+            WynntilsMod.info("Encoded item UTF-16: " + encodedItem.toUtf16String());
+        }
+
```

Comment:
I think you can remove this, most likely when debugging, we are going to use the chat setting.